src/parsec_client/operations/psa_algorithm.md

```diff
@@ -0,0 +1,347 @@
+   implementation does not provide. Implementations are encouraged to provide the modes that
+   applications need in preference to supporting direct access to ECB.
+- `CBC_NO_PADDING`: The Cipher Block Chaining (CBC) mode of a block cipher, with no padding. The
+   underlying block cipher is determined by the key type.
+- `CBC_PKCS7`: The Cipher Block Chaining (CBC) mode of a block cipher, with PKCS#7 padding. The
+   underlying block cipher is determined by the key type.
+
+#### Aead message
+
+Authenticated encryption with associated data (AEAD).
+
+| Field name | Type    | Description                                                                                     |
+|------------|---------|-------------------------------------------------------------------------------------------------|
+| `variant`  | `oneof` | Choice between a default length or a shortened tag algorithm, see below for the possible values |
+
+| `variant` oneof field name | Type                       |
+|----------------------------|----------------------------|
+| `aead_with_default_length` | `AeadWithDefaultLengthTag` |
+| `aead_with_shortened_tag`  | `AeadWithShortenedTag`     |
+
+##### AeadWithDefaultLengthTag enumeration
+
+- `AEAD_WITH_DEFAULT_LENGTH_TAG_NONE`: this default variant should not be used.
+- `CCM`: the CCM authenticated encryption algorithm. The underlying block cipher is determined by
+   the key type.
+- `GCM`: the GCM authenticated encryption algorithm. The underlying block cipher is determined by
+   the key type.
+- `CHACHA20_POLY1305`: the Chacha20-Poly1305 AEAD algorithm. The ChaCha20-Poly1305 construction is
+   defined in RFC 7539.
+
+##### AeadWithShortenedTag message
+
+AEAD algorithm with a shortened tag. An AEAD algorithm with a shortened tag is similar to the
+corresponding AEAD algorithm, but has an authentication tag that consists of fewer bytes. Depending
+on the algorithm, the tag length might affect the calculation of the ciphertext.
+
+| Field name   | Type                       | Description                                        |
+|--------------|----------------------------|----------------------------------------------------|
+| `aead_alg`   | `AeadWithDefaultLengthTag` | An AEAD algorithm                                  |
+| `tag_length` | `uint32`                   | Desired length of the authentication tag in bytes. |
+
+#### AsymmetricSignature message
+
+Asymmetric signature algorithms.
+
+| Field name | Type    | Description                                                       |
+|------------|---------|-------------------------------------------------------------------|
+| `variant`  | `oneof` | Asymmetric signature algorithm, see below for the possible values |
+
+| `variant` oneof field name | Type                    |
+|----------------------------|-------------------------|
+| `rsa_pkcs1v15_sign`        | `RsaPkcs1v15Sign`       |
+| `RsaPkcs1v15SignRaw`       | `rsa_pkcs1v15_sign_raw` |
```

Comment:
Something about the naming tells me these should've been the other way around

Reply:
Good catch! My `vim` macro failed me there